File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.py
def solution(s):
    answer = []
    temp = ''
    for i in range(len(s)):
        if temp.find(s[i]) == -1:
            temp += s[i]
            answer.append(-1)
        elif temp.find(s[i]) != -1:
            answer.append(i - temp.rfind(s[i]))
            temp += s[i]
            
            
    
    return answer

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    let dict = new Map();
    let arr = s.split("");
    let answer = [];
    for(let i=0; i<arr.length; i++){
        if (dict.has(arr[i])){
            answer.push(i-dict.get(arr[i]));
            dict.set(arr[i],i);
        }
        else{
            dict.set(arr[i],i);
            answer.push(-1);
        }
    }
    return answer;
}